ldes-fragmentisers/ldes-fragmentisers-substring/src/test/java/be/vlaanderen/informatievlaanderen/ldes/server/fragmentisers/substring/processor/SubstringPreProcessorTest.java

```diff
@@ -0,0 +1,111 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.substring.processor;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.tree.member.entities.Member;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.substring.config.SubstringConfig;
+import org.apache.jena.rdf.model.ModelFactory;
+import org.apache.jena.riot.Lang;
+import org.apache.jena.riot.RDFParserBuilder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.fragmentisers.substring.constants.SubstringConstants.ROOT_SUBSTRING;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+class SubstringPreProcessorTest {
+
+	private final SubstringConfig substringConfig = new SubstringConfig();
+	private final SubstringPreProcessor substringPreProcessor = new SubstringPreProcessor(substringConfig);
+
+	@Nested
+	class GetSubstringTarget {
+
+		@BeforeEach
+		void setUp() {
+			substringConfig.setFragmenterProperty("http://purl.org/dc/terms/description");
+		}
+
+		@Test
+		void shoudReturnValueFromFragmenterProperty_whenPresent() throws URISyntaxException, IOException {
+			final Member member = readLdesMemberFromFile();
+
+			String substringTarget = substringPreProcessor.getSubstringTarget(member);
+
+			assertEquals("omschríjvińĝ", substringTarget);
+		}
+
+		@Test
+		void shouldReturnNull_whenFragmenterPropertyNotPresentOnMember() {
+			Member member = new Member("id", ModelFactory.createDefaultModel(), new ArrayList<>());
+			assertNull(substringPreProcessor.getSubstringTarget(member));
+		}
+
+		private Member readLdesMemberFromFile()
+				throws URISyntaxException, IOException {
+			File file = new File(
+					Objects.requireNonNull(getClass().getClassLoader().getResource("example-ldes-member.nq")).toURI());
+
+			// noinspection resource
```

Comment:
Find this a confusing comment

Reply:
It is to block the warning of using 'Files.lines' in an unsafe way. In this test case it is OK but in non-test code it would not be OK.